batch/batch/front_end/inst_coll_selector.py

```diff
@@ -0,0 +1,82 @@
+import re
```

Comment:
I do not think this file and these classes enhance my reasoning capabilities about this code. PoolConfig seems like a stripped down Pool. PoolSelector seems very small? I thought all this would be managed by the InstanceCollectionManager who already has a list of pools and knows about the JPIM. It has all the information necessary to select a pool for a job.

Reply:
That code currently creates instances and starts the control loops which I didn't want to do for obvious reasons on the front end. I can try and refactor this again and see if I can solve it a different way.